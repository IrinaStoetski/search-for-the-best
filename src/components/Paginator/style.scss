@import '../../styles/shared/style.scss';

.pagination {
  @include flex-row-center;

  margin: 16px 0;
  overflow: hidden;

  @include max-width($landscape-tablet) {
    justify-content: start;
    max-width: 100%;
    overflow-x: auto;
  }

  li {
    @include flex-row-center;

    width: 36px;
    height: 36px;
    margin-right: 16px;
    padding: 8px 12px;
    border: 1px solid $primary-color;
    border-radius: 4px;
    cursor: pointer;

    a {
      color: $primary-color;
      font-size: 16px;
      text-decoration: none;
    }

    &.active {
      background-color: $primary-color;
      border: 1px solid transparent;

      a {
        color: #fff;
      }
    }
  }
}
