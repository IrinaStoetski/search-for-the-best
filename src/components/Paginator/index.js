import React, {useContext, useState} from 'react';
// ContextProviders
import { LayoutContext } from '../../contextProviders/LayoutContext';
//Components
import Pagination from "react-js-pagination"
//Styles
import './style.scss';

const Paginator = ({itemsCount, activePage}) => {
    const layoutContextData = useContext(LayoutContext);
    const {activeQuery, getRepositoriesList} = layoutContextData;


    const handlePageChange = (newPage) => {
        getRepositoriesList(activeQuery, newPage);
    }

    const amountArray = itemsCount / 30;
 
    return (
        <Pagination
        activePage={activePage}
        totalItemsCount={itemsCount / 30}
        pageRangeDisplayed={10}
        onChange={handlePageChange}
        hideDisabled
        hideFirstLastPages	
         />
    )
}

export default Paginator;