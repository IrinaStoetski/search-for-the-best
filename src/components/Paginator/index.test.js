/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import { LayoutContext } from '../../contextProviders/LayoutContext';
import Paginator from './index';

test('renders correctly Paginator component', () => {
  const getRepositoriesList = () => null;
  const isLoading = false;
  const activeQuery = {};

  const tree = renderer
    .create(
      <LayoutContext.Provider
        value={{
          getRepositoriesList,
          isLoading,
          activeQuery,
        }}
      >
        <Paginator />
      </LayoutContext.Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
