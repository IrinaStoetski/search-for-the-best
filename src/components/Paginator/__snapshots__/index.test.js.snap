// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator component renders correctly without props and test context provider 1`] = `null`;
